### ISSUE TITLE
Support Step Up with acr_values

### ISSUE THREAD

**Issue by Stillglade** [okta/okta-angular#125]
### Describe the feature request?

Okta supports the use of the acr_values parameter and the acr claim to implement step up authentication. It would be great if this library made this configuration easier to implement.

### New or Affected Resource(s)

Perhaps there is a way to extend the okta.guard or add a new guard which would allow a required acr claim value to be specified. Then during the canActivate call, it could validate the acr claim value from the token, and if not found it could trigger an updated login flow which contains the configured acr_values parameter in the login request.

### Provide a documentation link

[https://developer.okta.com/docs/guides/step-up-authentication/main/](https://developer.okta.com/docs/guides/step-up-authentication/main/)

**Comment by denysoblohin-okta** [comment]
Step Up MFA is supported in `okta-auth-js` since version `7.1.0`
You can pass `acrValues ` in [authorize options](https://github.com/okta/okta-auth-js/tree/master#authorize-options) for method [signInWithRedirect](https://github.com/okta/okta-auth-js/tree/master#signinwithredirectoptions) .
You can configure `OktaAuthModule ` by adding  `acrValues ` to `authConfig ` when constructing `OktaAuth ` instance.
Also `acr` claim is listed in type `UserClaims ` and is checked during token validation.

But supporting ACR values in `okta-angular` in a dedicated guard seems reasonable feature request.

Internal ref: OKTA-597151

**Comment by Stillglade** [comment]
One solution would be to update the existing Guard. For example, the new usage would look something like this:

```// myApp.module.ts

import {
  OktaAuthGuard,
  ...
} from '@okta/okta-angular';

const appRoutes: Routes = [
  {
    path: 'protected',
    component: MyProtectedComponent,
    canActivate: [ OktaAuthGuard ],
    data: {acrValues: "<from predefined list of values>"},
    children: [{
      // children of a protected route are also protected
      path: 'also-protected'
    }]
  },
  ...
]
```

This would then be accessible within the okta.guard.ts methods (canActivate and canLoad) using `route.data.acrValues` and could be passed down into the existing handleLogin method as an optional second parameter.

**Comment by denysoblohin-okta** [comment]
This feature will be released in upcoming 6.3.0
